DigiMeSDK/Core/Classes/Errors/NSError+API.m

```diff
@@ -10,4 +10,58 @@
 
 @implementation NSError (API)
 
++ (NSError *)apiErrorWithReason:(NSString * _Nullable)errorReason reference:(NSString * _Nullable)errorReference
+{
+    NSString *message = [@[[[self class] errorReason:errorReason], [[self class] errorReference:errorReference]] componentsJoinedByString:@","];
+    return [NSError errorWithDomain:DME_API_ERROR code:[[self class] errorCode:errorReason] userInfo:@{ NSLocalizedDescriptionKey:message}];
+}
+
++ (NSInteger)errorCode:(NSString * _Nullable)reasonErrorMessage
```

Comment:
To clarify my earlier comments, I don;t think we need to pass and process `reason` in this way. The app currently always sends the `failedToRetrieveContract` error back, so we could just add an extra case in the AuthError enum to signify that authorization has failed in the the app, and attach the `reference`.

If we wanted to go down the route of passing `reason` and `code` in the future, there is certainly a better way of doing it, eg:
`code=404&reason=xyz`.

Please remove the regex parsing of the above, leaving behind only `reference` with predefined error code.